### PR TITLE
[D] Separate built-in type-aliases from built-in types

### PR DESCRIPTION
**Pull request by ichordev**
Changes:
- Separated built-in type-aliases (e.g. `size_t`, `string`) from built-in types (e.g. `int`, `float`) so that their identifiers aren't treated as reserved (they aren't, e.g. you can make your own `string`), and so that they can be highlighted separately.

### CODE REVIEW

**Comment by ichordev** [comment]
I swear it works this time.

**Review by keith-hall — APPROVED** [review]

File: D/D.sublime-syntax [hunk 1]
```diff
@@ -2234,15 +2238,19 @@ contexts:
...
-          pop: true
-    - include: not-whitespace-illegal-pop
+      set: type-identifier-after
+    - include: not-whitespace-illegal-pop
+  type-identifier-after:
+      - match: \.
+        scope: meta.path.d punctuation.accessor.dot.d
+        set: type-identifier
+      - match: '(?=\S)'
+        pop: true
```
Comment (deathaxe) [inline D/D.sublime-syntax:2253]: Overindented

```suggestion
    - match: \.
      scope: meta.path.d punctuation.accessor.dot.d
      set: type-identifier
    - match: '(?=\S)'
      pop: true
```

File: D/D.sublime-syntax [hunk 2]
```diff
@@ -2234,15 +2238,19 @@ contexts:
     - match: '\b{{name}}\b(?=\s*!)'
       scope: meta.function-call.template.d meta.path.d variable.function.d
       pop: true
+    - match: '\b{{builtin_type_alias}}\b'
+      scope: meta.path.d storage.type.d support.type.builtin-alias.d
```
Comment (deathaxe) [inline D/D.sublime-syntax:2242]: As above, just `support.type` would be enough.

```suggestion
      scope: meta.path.d support.type.builtin-alias.d
```

**Review by deathaxe — APPROVED** [review]

**Comment by deathaxe** [comment]
Please fix the failing test. Otherwise LGTM!

**Review by deathaxe — APPROVED** [review]

**Review by keith-hall — APPROVED** [review]

**Pull request merged**